### ISSUE TITLE
RBAC rules for operator?

### ISSUE THREAD

**Issue by adarshaj** [zalando-incubator/postgres-operator#233]
What's a good set of rbac ruleset for operator?
After tweaking a bit, I still see the following in logs of operator pod - 

```
 github.com/zalando-incubator/postgres-operator/pkg/controller/controller.go:232: Failed to list *spec.Postgresql: unknown (get postgresqls.acid.zalan.do
```

**Comment by alexeyklyukin** [comment]
Hi,

I think we just run the operator under the cluster-admin role, but my first take on the RBAC would be something like this: 

https://github.com/zalando-incubator/postgres-operator/pull/234/

The error you've received comes from the need of the operator to define and read CRD objects.
This is covered by the
```
rules:
- apiGroups:
  - acid.zalan.do
  resources:
  - postgresqls
  verbs:
  - "*"
- apiGroups:
  - apiextensions.k8s.io
  resources:
  - customresourcedefinitions
  verbs:
  - "*"
```

**Comment by adarshaj** [comment]
I had a cluster role, so CRDs wasn't a problem, but I'm not sure why I see the above message. I think I'll start from scratch again, I might have messed up changes.

**Comment by mobilesfinks** [comment]
Add ClusterRole operator  with next rules:
```apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: operator
rules:
- apiGroups:
  - ""
  resources:
  - configmaps
  verbs:
  - create
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - ""
  resources:
  - endpoints
  verbs:
  - get
  - patch
  - update
  - create
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - pods
  verbs:
  - get
  - list
  - patch
  - update
  - watch
- apiGroups:
  - ""
  resources:
  - nodes
  verbs:
  - get
  - list
- apiGroups:
  - acid.zalan.do
  resources:
  - postgresqls
  verbs:
  - '*'
- apiGroups:
  - apiextensions.k8s.io
  resources:
  - customresourcedefinitions
  verbs:
  - '*'
```

But operator cannot get nodes and i see error in log


> E0226 14:37:17.708507       1 reflector.go:201] github.com/zalando-incubator/postgres-operator/pkg/controller/controller.go:242: Failed to list *v1.Node: nodes is forbidden: User "system:serviceaccount:inner:operator" cannot list nodes at the cluster scope

A temporary solution is to expose clusterrolebinding on the cluster-admin
So everything started, but this is an incorrect solution


How can i fix this problem?

**Comment by alexeyklyukin** [comment]
```
- apiGroups:
  - ""
  resources:
  - nodes
  verbs:
  - get
  - list
```
You also need watch on the nodes (see https://github.com/zalando-incubator/postgres-operator/blob/1406581c7fee7ea3d80c80e062e10abebd3c293e/manifests/operator-rbac.yaml)

It's unusual that the error mentions complains about failing to list nodes and your list privileges are there. Did you use ``ClusterRoleBinding`` to bind your ``ClusterRole`` with the ``serviceAccount``?

**Comment by mobilesfinks** [comment]
yes, i use `ClusterRoleBinding` to bind `ClusterRole` with the `serviceAccount`
Add watch verb - it seems work
```
- apiGroups:
  - ""
  resources:
  - nodes
  verbs:
  - get
  - list
  - watch
```
Thank you. 
Take yours `ClusterRole` definition

**Comment by zerg-junior** [comment]
closed by #344 

this PR adds example RBAC rules in `/manifests` and an example of a role binding in the controller initialization code
